### PR TITLE
smcp v1 and v2 mesh tests

### LINKED ISSUE #443: Move test-specific SMCP creation and setup into test code
Currently "make install-mesh" contains test-specific setup  (certificate generation and specific ingress configuration in SMCP).

We should change this so that
* new target "make install-mesh-operator" that installs just the servicemesh operator (and no SMCP)
* "make install-mesh" installs a generic SMCP that would not be used in tests.
* mesh tests will use their own test namespaces for their own test SMCP and a different test namespace as the SMMR member.
* mesh tests will generate their certificates in test code.

This will allow mesh test to each have their own customized SMCP and we won't have tight dependencies between test and "make install-mesh" .
The downside is that mesh test will take a little bit longer, as they will each wait for their SMCP readiness.

### PR DESCRIPTION
**Pull request by maschmid**
drop mesh scripts, move setup logic to tests

fixes #443

### CODE REVIEW

**Comment by maschmid** [comment]
hm, weird, kiali operator didn't install, from the olm-operator logs
```
2020-12-07T12:30:05.990009064Z time="2020-12-07T12:30:05Z" level=info msg="couldn't ensure RBAC in target namespaces" csv=kiali-operator.v1.24.2 error="no owned roles found" id=5lI/p namespace=openshift-operators phase=Pending
2020-12-07T12:30:05.990037015Z E1207 12:30:05.990026       1 queueinformer_operator.go:290] sync {"update" "openshift-operators/kiali-operator.v1.24.2"} failed: no owned roles found
```

**Comment by maschmid** [comment]
ServiceMesh tests passing in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/726/pull-ci-openshift-knative-serverless-operator-master-4.6-operator-e2e-aws-ocp-46/1335961934180126720/artifacts/operator-e2e-aws-ocp-46/operator-e2e/build-WAUJd0wp/junit_JT48ZEaz.xml 

I'll remove the https://github.com/openshift-knative/serverless-operator/pull/726/commits/4eb2c13352dcaec8ed31c7577b27b5a631c4013c , as we don't want to run them in all configurations (install-mesh is enabled only in continuous jobs)

**Comment by maschmid** [comment]
/unhold

(note, Service Mesh is not installed with 4.6-operator-e2e-aws-ocp-46 , so the new tests will be skipped in this run )

**Comment by nak3** [comment]
Thank you for this great work! It looks good to me overall.

There is one discussion I think. I think that we can use Manifestival for the resource manipulation. For example, if we want to create serviceentry:

```
  patch := "test/some_path/serviceentry.yaml"
  manifest, err := mf.NewManifest(path, mf.UseClient(mfclient.NewClient(client)))
  if err != nil {
   return err
  }

  if err := manifest.Apply(); err != nil {
   return err
  }
```

Does it make more readable code? I feel that current resources in the code is a little bit difficult to read. WDYT?
(It will need an entire change so I am fine that we can merge this PR now and consider about it later.)

**Comment by nak3** [comment]
@maschmid Friendly reminder. Any update on this?

**Comment by maschmid** [comment]
@nak3 Thanks for the review! (and sorry for the long delay... )

As for using the files via Manifestival I am not convinced... In some/most cases we need some kind of templating and setting values inside a golang structure seems most readable in such cases (instead of manipulating a structure that comes from a completely different file).. (and mixing different approaches for static vs. templated structures doesn't feel nice either).

File: test/servicemesh.go [hunk 1]
```diff
@@ -4,12 +4,100 @@ import (
...
+	}
+}
+
+func AddServiceMeshControlPlaneV1IngressGatewaySecretVolume(smcp *unstructured.Unstructured, name, secretName, mountPath string) error {
+	secretVolume := make(map[string]interface{})
+	secretVolume["name"] = name
+	secretVolume["secretName"] = secretName
+	secretVolume["mountPath"] = mountPath
+
+	secretVolumes, found, _ := unstructured.NestedSlice(smcp.Object, "spec", "istio", "gateways", "istio-ingressgateway", "secretVolumes")
```
Comment (nak3) [inline test/servicemesh.go:57]: The error from `NestedSlice` is ignored by design? If so, could you add the comment?

File: README.md [hunk 2]
```diff
@@ -265,6 +251,31 @@ spec:
   - Ingress
 ```
 
+Then, create Knative Service with `sidecar.istio.io/inject: "true"` annotation in your namespace,
+which is one of the namespaces in the `ServiceMeshMemberRoll`.
+
+```sh
+cat <<EOF | oc apply -n default -f -
```
Comment (nak3) [inline README.md:258]: nit:

```suggestion
cat <<EOF | oc apply -n $NAMESPACE_YOU_WANT_TO_ADD -f -
```

**Review by nak3 — COMMENTED** [review]
Thank you. Commented about a few nit.

File: test/servicemesh.go (hunk 1 above)
Comment (maschmid) [reply test/servicemesh.go]: Fixed a few more of those as well, thanks!

**Comment by nak3** [comment]
/assign @markusthoemmes @mgencur
for `/approve`

**Review by markusthoemmes — APPROVED** [review]

**Comment by markusthoemmes** [comment]
/retest

Infra issues.

**Pull request merged**